gwt-window/src/main/java/org/gwtproject/user/window/client/Cookies.java

```diff
@@ -187,7 +186,7 @@ public static void setUriEncode(boolean encode) {
   private static void loadCookies(HashMap<String, String> m) {
     String docCookie = document.cookie;
     if (Js.isTruthy(docCookie) && !docCookie.isEmpty()) {
-      String[] crumbs = Js.<JsString>cast(docCookie).split("; ");
+      String[] crumbs = Js.<String>cast(docCookie).split("; ");
```

Comment:
That cast from `String` to `JsString` was intended (in the current gwtproject/gwt-http), in order to uses the JS `split()` behavior rather than the Java one, emulated in javaemul's `java.lang.String`. The Java behavior, while probably not a problem here, is a lot more code.

The cast now turns a `String` into a `String`, which is completely useless.

Reply:
The `Js.<JsString>cast` will fail in j2cl. This was the first intend to solve the problem and it works. The latest change is to use `Js.<JsString>uncheckedCast` which does not fail in J2CL.

Reply:
> The cast now turns a `String` into a `String`, which is completely useless.

Agree. As @FrankHossfeld mentioned `Js.<JsString>` causes the CCE, so `Js.<JsString>uncheckedCast` should solve both issues.